### DISCUSSION TITLE
(Optimization) Don't spawn abyss islands if there's no abyss entrance

### DISCUSSION THREAD

**Discussion by girlyguppy** [FakeFishGames/Barotrauma#15349] [Ideas and suggestions]
![image](https://github.com/user-attachments/assets/db2a0bcf-8d7f-4eb7-a0a5-2d63f74e89dc)
![image](https://github.com/user-attachments/assets/88ade850-8b9e-4582-b973-52cd98b394a7)

this is an optimization because how could you get to these abyss islands if there's no entrance to the abyss? you can't, so why have all those caves in the level at all

**Comment by SomeRandomNoobKekeke** [comment]
`AbyssIslandCount="0"` in `LevelGenerationParameters`?
![Dev from below](https://github.com/user-attachments/assets/2029441d-5796-4739-b1fc-ecfc243a7561)

**Reply by girlyguppy** [reply]
yeah, but then if there *is* an entrance, they won't spawn :barodev:
